bindings/nodejs/src/functions/uart.cc

```diff
@@ -0,0 +1,176 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <sol-uart.h>
+#include <node.h>
+#include <nan.h>
+
+#include "../common.h"
+#include "../hijack.h"
+#include "../structures/sol-js-uart.h"
+#include "../structures/js-handle.h"
+
+using namespace v8;
+
+class SolUART : public JSHandle<SolUART> {
+public:
+    static const char *jsClassName() { return "SolUART"; }
+};
+
+static void sol_uart_read_callback(void *user_data, struct sol_uart *uart,
+    unsigned char byte_read) {
+    Nan::HandleScope scope;
+    sol_uart_data *uart_data = (sol_uart_data *)user_data;
+    Nan::Callback *callback = uart_data->rx_cb;
+    if (!callback)
+        return;
+
+    Local<Value> arguments[2] = {
+        SolUART::New(uart_data),
```

Comment:
I'm not sure it's such a good idea to create a new ```SolUART``` here, because you're storing the pointer ```uart_data``` in yet another object. Now you have the same pointer (given to you by soletta in ```sol_uart_open()```) in two JS objects. You invalidate one of them in ```bind_sol_uart_close()``` by doing ```Nan::SetInternalFieldPointer()```, but you never invalidate this object the same way. It might be better to store a persistent reference to the SolUART object you create in ```bind_sol_uart_open()``` in the structure ```uart_data``` and pass it to the JS callback here. That way, there's only one JS object corresponding to the ```sol_uart *``` in play.